src/vs/workbench/contrib/chat/browser/chatListRenderer.ts

```diff
@@ -534,7 +536,7 @@ export class ChatAccessibilityProvider implements IListAccessibilityProvider<Cha
 		}
 
 		if (isResponseVM(element)) {
-			return element.response.value;
+			return this._getLabelWithCodeBlockCount(element);
```

Comment:
Can you pull this info off of `codeBlocksByResponseId` instead of storing it on the viewmodel object? You can expose that publicly on the renderer. I don't want to store the same info in two places and I want to keep the viewModels as thin as possible.

Reply:
Ok just added something which updatees the aria label after the render, but it's still not working and seems to use the aria label that gets set originally

Reply:
is there a way to trigger an update of the `ariaLabel` that i'm missing?

Reply:
I think that this was working with my original change because changing `element.codeBlockCount` was triggering `onDidChangeItems` and maybe it was re-rendering?

https://github.com/microsoft/vscode/blob/a69ee6a02988f66332fdf362ab710999d3b94819/src/vs/workbench/contrib/chat/browser/chatWidget.ts#L163

Reply:
> I think that this was working with my original change because changing element.codeBlockCount was triggering onDidChangeItems

I don't see why that would happen?

Reply:
I don't think that there is any guarantee that the accessibility provider will be called before an element is rendered. You can look at the list code or ask Joao to see whether that's the case or if there's any way to customize that. 

If not, that's probably an issue for us. But what we can maaaybe do is cache some info about rendered markdown on the viewmodel, and be able to kick that off either from the accessibilityProvider or the renderer. I want to make sure we don't do the same work twice because that is the most expensive part of rendering this list. But it would complicate things quite a bit.

The last resort solution could be evil hacks where we just set the aria-label manually when the renderer is called, but that requires reaching out of the content element up to the monaco-list-row and we shouldn't really be manipulating that element from a renderer.